network/hostnetwork/future/handler.go

```diff
@@ -98,5 +98,5 @@ func shouldProcessPacket(future Future, p *packet.Packet) bool {
 	typesShouldBeEqual := p.GetType() == future.Request().GetType()
 	responseIsForRightSender := future.Receiver().Equal(*p.Sender)
 
-	return typesShouldBeEqual && (responseIsForRightSender || p.GetType() == types.Ping)
+	return typesShouldBeEqual && (responseIsForRightSender || p.GetType() == types.Ping || p.GetType() == types.Replication)
```

Comment:
do we really need a new type of the packets?

Reply:
fixed